src/communication.ts

```diff
@@ -98,8 +98,14 @@ export class Communication {
     }
 
     // function that adds a new subscriber, which is called every time a message is received
-    subscribeMessage(subscriber: Subscriber): void {
-        this.socketServer.on('message', subscriber);
+    subscribeMessage(subscriber: Subscriber, type?: string): void {
+        //this.socketServer.on('message', subscriber);
+
+        this.socketServer.on('message', (message) => {
+            if (!type || message.getType() === type) {
+                subscriber(message);
```

Comment:
when i convert message to message object external user testcase failed but with this condition it's working @theblockstalk @Rebal67

Reply:
message is of type string 
string doesnt have message,getType function
this should be encoded to a Message object